src/error.rs

```diff
@@ -388,6 +388,24 @@ impl From<std::io::ErrorKind> for ErrorKind {
     }
 }
 
+#[cfg(feature = "openssl-tls")]
```

Comment:
I think we'll want to match the error types that the existing rustls implementaiton uses as best we can here. So for example, any errors returned as part of `make_ssl_connector` and the subsequent configuration seem to map nicely to `InvalidTlsConfig`, and everything after that would be a custom `std::io::Error`, since that appears to be [what `rustls` does (for better or worse)](https://docs.rs/tokio-rustls/0.23.2/src/tokio_rustls/lib.rs.html#127). I'm not sure if a single `From` implementation will be able to handle this correctly unfortunately, so we may have to do it at the OpenSSL call sites. 

More generally, `ErrorKind::Authentication` has thus far been reserved for MongoDB authentication failures (including x509 auth failures, which should be fine without this `From` I believe), not TLS verification failures.